DC-575: Customizable column headers in data catalog list view

pshapiro4broad opened the pull request:
Add UI support for customizing which columns are visible and what order they appear in.

Use the cog icon to open up the Select columns dialog.

![image](https://user-images.githubusercontent.com/1799360/202485453-7930904d-e2af-409e-8851-52925f2b9b65.png)

The dialog lets users enable/disable columns and drag to re-order them.

![image](https://user-images.githubusercontent.com/1799360/202485577-9d980e55-aa1f-4f51-9b6a-4f8d300063a5.png)

After customizing to include all columns, and with order changed.

<img width="1635" alt="image" src="https://user-images.githubusercontent.com/1799360/202485974-fe2e6a28-0f33-4080-a5ef-f7f1265d6edf.png">


For code reviewers: there's a big whitespace change, using Hide whitespace in the GitHub diff will make it clearer what was changed.

pshapiro4broad committed:
DC-575: WIP, custom column headers

s-rubenstein committed:
Massage the data for columns

pshapiro4broad committed:
Add a few more columns that aren't visible by default, add some comments.

pshapiro4broad committed:
Merge branch 'dev' into ps-DC-575-catalog-custom-column-headers

s-rubenstein commented on src/components/table.js:
```diff
@@ -916,6 +916,7 @@ export const ColumnSettings = ({ columnSettings, onChange }) => {
  * @param {Object[]} columnSettings - current column list, in order
  * @param {string} columnSettings[].name
  * @param {bool} columnSettings[].visible
+ * @param modalWidth - Width of the selector modal
```
I added this, but I think the format is wrong. Rather than the variable name, it should be the type.

pshapiro4broad replied:
I've fixed this and the rest of the params, since they weren't correct either -- since they're destructured parameters you have to use an object prefix for the param name to make the jsdoc work.

s-rubenstein reviewed (commented):
I wonder if we could unit test this. At the very least, I’d suggest pulling the functions to map the columnKey array out into helpers, and then unit testing those, because that might serve as useful documentation for their functionality.

pshapiro4broad committed:
fix destructured parameter documentation

pshapiro4broad committed:
Merge branch 'dev' into ps-DC-575-catalog-custom-column-headers

pshapiro4broad committed:
Refactor out conversion functions and add unit tests

pshapiro4broad committed:
Merge branch 'dev' into ps-DC-575-catalog-custom-column-headers

pshapiro4broad committed:
full stop

pshapiro4broad committed:
Merge remote-tracking branch 'origin/ps-DC-575-catalog-custom-column-headers' into ps-DC-575-catalog-custom-column-headers

pshapiro4broad commented:
@s-rubenstein 
> I wonder if we could unit test this. At the very least, I’d suggest pulling the functions to map the columnKey array out into helpers, and then unit testing those, because that might serve as useful documentation for their functionality.

I refactored out these functions and added unit tests.

pshapiro4broad committed:
added a comment

pshapiro4broad committed:
Merge branch 'dev' into ps-DC-575-catalog-custom-column-headers

s-rubenstein reviewed (approved):
Lgtm

pshapiro4broad committed:
Merge branch 'dev' into ps-DC-575-catalog-custom-column-headers

pshapiro4broad committed:
Merge branch 'dev' into ps-DC-575-catalog-custom-column-headers

pshapiro4broad committed:
Merge branch 'dev' into ps-DC-575-catalog-custom-column-headers

pshapiro4broad committed:
Merge branch 'dev' into ps-DC-575-catalog-custom-column-headers

pshapiro4broad committed:
Merge branch 'dev' into ps-DC-575-catalog-custom-column-headers

lu-c reviewed (commented):
LGTM :D

pshapiro4broad committed:
Merge branch 'dev' into ps-DC-575-catalog-custom-column-headers

Abulu123 reviewed (approved):
took a look and looks good for what I understand
